librz/core/cmd_debug.c

```diff
@@ -5036,6 +4410,655 @@ RZ_IPI int rz_cmd_debug(void *data, const char *input) {
+	RzList *list = rz_debug_frames(core->dbg, UT64_MAX);
+	if (!list) {
+		return RZ_CMD_STATUS_ERROR;
+	}
+	asciiart_backtrace(core, list);
+	rz_list_free(list);
+	return RZ_CMD_STATUS_OK;
+}
+
+// dbte
+RZ_IPI RzCmdStatus rz_cmd_debug_bt_enable_bp_trace_handler(RzCore *core, int argc, const char **argv) {
+	for (int i = 1; i < argc; i++) {
+		if (!strcmp(argv[i], "*")) {
+			if (!rz_bp_set_trace_all(core->dbg->bp, true)) {
+				RZ_LOG_ERROR("Failed to enable all breakpoints' trace\n");
+			}
+			break;
+		} else {
+			ut64 addr = rz_num_math(core->num, argv[i]);
+			if (!rz_bp_set_trace(core->dbg->bp, addr, true)) {
+				RZ_LOG_ERROR("Failed to enable trace for breakpoint at 0x%" PFMT64x "\n", addr);
+			}
+		}
+	}
+	return RZ_CMD_STATUS_OK;
+}
+
+// dbtd
+RZ_IPI RzCmdStatus rz_cmd_debug_bt_disable_bp_trace_handler(RzCore *core, int argc, const char **argv) {
+	for (int i = 1; i < argc; i++) {
+		if (!strcmp(argv[i], "*")) {
+			if (!rz_bp_set_trace_all(core->dbg->bp, false)) {
+				RZ_LOG_ERROR("Failed to enable all breakpoints' trace\n");
+			}
+			break;
+		} else {
+			ut64 addr = rz_num_math(core->num, argv[i]);
+			if (!rz_bp_set_trace(core->dbg->bp, addr, false)) {
+				RZ_LOG_ERROR("Failed to enable trace for breakpoint at 0x%" PFMT64x "\n", addr);
+			}
+		}
+	}
+	return RZ_CMD_STATUS_OK;
+}
+
+// dbts
+RZ_IPI RzCmdStatus rz_cmd_debug_bt_toggle_bp_trace_handler(RzCore *core, int argc, const char **argv) {
+	for (int i = 1; i < argc; i++) {
+		if (!strcmp(argv[i], "*")) {
+			RzListIter *iter;
+			RzBreakpointItem *bpi;
+			rz_list_foreach (core->dbg->bp->bps, iter, bpi) {
+				bpi->trace = !bpi->trace;
+			}
+		} else {
+			ut64 addr = rz_num_math(core->num, argv[i]);
+			RzBreakpointItem *bpi = rz_bp_get_in(core->dbg->bp, addr, 0);
+			if (!bpi) {
+				RZ_LOG_ERROR("No breakpoint found at 0x%" PFMT64x "\n", addr);
+			} else {
+				bpi->trace = !bpi->trace;
+			}
+		}
+	}
+	return RZ_CMD_STATUS_OK;
+}
+
+// dbx
+RZ_IPI RzCmdStatus rz_cmd_debug_bp_set_expr_cur_offset_handler(RzCore *core, int argc, const char **argv) {
+	if (argc == 1) {
+		RzBreakpointItem *bpi;
+		RzListIter *iter;
+		rz_list_foreach (core->dbg->bp->bps, iter, bpi) {
+			rz_cons_printf("0x%08" PFMT64x " %s\n", bpi->addr, rz_str_get(bpi->expr));
+		}
+	} else if (argc == 2) {
+		RzBreakpointItem *bpi = rz_bp_get_at(core->dbg->bp, core->offset);
+		if (!bpi) {
+			RZ_LOG_ERROR("No breakpoint found at current offset: 0x%" PFMT64x "\n", core->offset);
+			return RZ_CMD_STATUS_ERROR;
+		}
+		if (!rz_bp_item_set_expr(bpi, argv[1])) {
+			RZ_LOG_ERROR("Failed to set expression\n");
+			return RZ_CMD_STATUS_ERROR;
+		}
+	}
+	return RZ_CMD_STATUS_OK;
+}
+
+// dbw
+RZ_IPI RzCmdStatus rz_cmd_debug_add_watchpoint_handler(RzCore *core, int argc, const char **argv) {
+	if (argc == 3) {
+		int hwbp = rz_config_get_i(core->config, "dbg.hwbp");
+		add_breakpoint(core, argv[1], argv[2], hwbp, true);
+	} else {
```

Comment:
No need for `RZ_CMD_STATUS_WRONG_ARGS` - the rzshell handles that automatically, according to the number of the arguments defined in the YAML file.